diff --git a/docroot/.htaccess b/docroot/.htaccess
index 7c93ef41..37fce10d 100644
--- a/docroot/.htaccess
+++ b/docroot/.htaccess
@@ -116,8 +116,6 @@ DirectoryIndex index.php index.html index.htm
 
   # Pass all requests not referring directly to files in the filesystem to
   # index.php. Clean URLs are handled in drupal_environment_initialize().
-  # Allow access to simplesaml paths
-  RewriteCond %{REQUEST_URI} !^/simplesaml
   RewriteCond %{REQUEST_FILENAME} !-f
   RewriteCond %{REQUEST_FILENAME} !-d
   RewriteCond %{REQUEST_URI} !=/favicon.ico
